### PR TITLE
Developer-friendly ``pymc3.distributions``

### PR DESCRIPTION
**Pull request by superbobry**
This is a purely cosmetic PR with a goal of making `pymc3.distributions` more readable from the developer point of view. If the idea sounds good, I'll proceed to the rest of the modules.

A related issue is a lack of API docs and (as a consequence) malformed or non-informative docstrings.

### CODE REVIEW

File: pymc3/distributions/discrete.py [hunk 1]
```diff
@@ -203,53 +199,54 @@ class NegativeBinomial(Discrete):
...
-                      mu > 0, alpha > 0, value >= 0)
-
-        return switch(1*(alpha > 1e10),
-                      pois,
-                      negbinom)
+        # Return Poisson when alpha gets very large.  This only
+        # works because both Poisson and NegativeBinomial use the
+        # same attribute names.
+        return T.switch(1 * (alpha > 1e10),
+                        Poisson.logp(self, value),
```
Comment (fonnesbeck) [inline pymc3/distributions/discrete.py]: You probably need to use the `Poisson.dist` classmethod here

**Comment by fonnesbeck** [comment]
These look great. Other than the unit test failure, I don't have a problem with these changes.

**Comment by twiecki** [comment]
Yes, totally agree.

**Comment by superbobry** [comment]
I'm sorry, this was too early to merge. The PR breaks `ScalarShape.test_categorical` in `test_distributions_random.py`. I will add a fix, can you please revert it for now?

Also, is there a reason for only running some of the tests on Travis?

**Update**: this line in `.travis.yml` makes nose skip `test_distributions_random.py`:

``` yaml
env:
  - TESTCMD=" -vv --with-timer -e test_examples -e test_distributions"
```

**Comment by fonnesbeck** [comment]
Yeah, this was not ready to merge yet. I will roll it back.

**Comment by fonnesbeck** [comment]
Reverted. I think you will have to open another PR; I don't think I can reopen this one.

**Comment by fonnesbeck** [comment]
@superbobry regarding tests, we were having serious issues getting all of the tests run in a reasonable amount of time on Travis. Those lines may have been added to cut testing time. Not sure if they are needed anymore.

**Comment by twiecki** [comment]
Sorry about that!

**Pull request merged**